R-package/demo/leaf_stability.R

```diff
@@ -180,20 +181,20 @@ new_data3 <- data.frame(
             )
             , 1e-15
         )
-        , 1 - 1e-15
+        , 1.0 - 1e-15
     )
 )
-new_data3$Z <- -1 * (agaricus.test$label * log(new_data3$Y) + (1 - agaricus.test$label) * log(1 - new_data3$Y))
+new_data3$Z <- -1.0 * (agaricus.test$label * log(new_data3$Y) + (1L - agaricus.test$label) * log(1L - new_data3$Y))
 new_data3$binned <- .bincode(
     x = new_data3$X
     , breaks = quantile(
         x = new_data3$X
-        , probs = (1:9) / 10
+        , probs = (1L:9L) / 10.0
```

Comment:
`seq_len(9L)` for safety

Reply:
could you explain how `seq_len(9L)` is safer than `1L:9L`?

Reply:
I'm happy to change these in this PR for consistency with the other code where we initialize vectors like this, but I don't think there is a difference in safeness since `1L` and `9L` are both integer literals.

Reply:
changed these

Reply:
@jameslamb It's just in case if those are modified in the future, the probability vector would throw an error if its value is negative (quantile already throws another kind of error message with negative probs).

We also usually don't expect quantile (as users) to take the 0 percentile (minimum), it might happen though (and that would have to be explicit).